javaslang/src/main/java/javaslang/collection/Array.java

```diff
@@ -899,38 +898,12 @@ public int length() {
 
     @Override
     public Array<T> removeAll(T element) {
-        final java.util.List<T> list = new ArrayList<>();
-        for (int i = 0; i < length(); i++) {
-            T value = get(i);
-            if (!element.equals(value)) {
-                list.add(value);
-            }
-        }
-        if (list.size() == length()) {
-            return this;
-        } else {
-            return wrap(list.toArray());
-        }
+        return (Array<T>) IndexedSeq.super.removeAll(element);
```

Comment:
cast is needed, as return value is the same as the subclass, which isn't available in the parent class.
Also, in case of trait-like inheritance, it seems that only the directly declared parents are accessible (hence all the delegating methods)